profiler/src/ProfilerEngine/Datadog.Profiler.Native/ClrEventsParser.cpp

```diff
@@ -218,17 +218,40 @@ ClrEventsParser::ParseGcEvent(uint64_t timestamp, DWORD id, DWORD version, ULONG
     {
         // This event provides the size of each generation after the collection
         // --> not used today but could be interesting to detect leaks (i.e. gen2/LOH/POH are growing)
+        uint64_t gen2Size = 0;
+        uint64_t lohSize = 0;
+        uint64_t pohSize = 0;
 
-        // TODO: check for size and see if V2 with POH numbers could be read from payload
-        GCHeapStatsV1Payload payload = {0};
-        ULONG offset = 0;
-        if (!Read<GCHeapStatsV1Payload>(payload, pEventData, cbEventData, offset))
+        // check for size and see if V2 with POH numbers could be read from payload
+        if (version == 1)
         {
-            return;
+            GCHeapStatsV1Payload payload = {0};
+            ULONG offset = 0;
+            if (!Read<GCHeapStatsV1Payload>(payload, pEventData, cbEventData, offset))
+            {
+                return;
+            }
+
+            gen2Size = payload.GenerationSize2;
+            lohSize = payload.GenerationSize3;
+            pohSize = 0;
+        }
+        else if (version == 2)
+        {
+            GCHeapStatsV2Payload payload = {0};
+            ULONG offset = 0;
+            if (!Read<GCHeapStatsV2Payload>(payload, pEventData, cbEventData, offset))
+            {
+                return;
+            }
+
+            gen2Size = payload.GenerationSize2;
+            lohSize = payload.GenerationSize3;
+            pohSize = payload.GenerationSize4;
         }
```

Comment:
shouldn't we add something that would warn us when a new version of the event is received ?